src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_kernel_emitter.cpp

```diff
@@ -0,0 +1,335 @@
+    validate_arguments(in, out);
+    emit_impl(in, out);
+}
+
+void jit_kernel_emitter::validate_arguments(const std::vector<size_t> &in, const std::vector<size_t> &out) const {
+    OV_CPU_JIT_EMITTER_ASSERT(in.empty() && out.empty(), ": Expects 0 registers on input and output");
+    const auto num_params = num_inputs + num_outputs + num_unique_buffers;
+    // The number of used gpr may be >= num_params since LoopBegin+LoopEnd could also use gpr to store work_amount
+    OV_CPU_JIT_EMITTER_ASSERT(data_ptr_regs_idx.size() == num_params,
+                              "Number of inputs and outputs is inconsistent with the number of allocated registers ", num_params,
+                              " data_ptr_regs_idx.size() = ", data_ptr_regs_idx.size());
+}
+
+void jit_kernel_emitter::init_body_regs(const std::set<size_t>& kernel_regs,
+                                        const std::vector<size_t> &pool_vec_idxs, const std::vector<size_t> &pool_gpr_idxs) {
+    // Initialize pools of gp and vec registers
+    // Reserve kernel regs (reg_indexes_idx and, if there is, reg_runtime_params_idx), since they'll be used to pass runtime call args to kernel
+    init_reg_pools(kernel_regs, {});
+
+    mapping_info gpr_map_pool({}, gp_regs_pool);
+    mapping_info vec_map_pool({}, vec_regs_pool);
+
+    // Note that we can't use kernel_regs to store data pointers because
+    // these regs are used to calculate offsets for the data pointers
+    map_abstract_registers(gpr_map_pool, vec_map_pool, mem_access_exprs);
+    for (const auto& abstract_to_physical : gpr_map_pool.first)
+        data_ptr_regs_idx.push_back(abstract_to_physical.second);
+
+    vec_map_pool.second.insert(vec_map_pool.second.end(), pool_vec_idxs.cbegin(), pool_vec_idxs.cend());
+    gpr_map_pool.second.insert(gpr_map_pool.second.end(), pool_gpr_idxs.cbegin(), pool_gpr_idxs.cend());
+    map_abstract_registers(gpr_map_pool, vec_map_pool, general_exprs);
+}
+
+void jit_kernel_emitter::emit_impl(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
+    h->preamble();
+
+    auto data_ptr_regs = transform_idxs_to_regs(data_ptr_regs_idx);
+
+    init_data_pointers(data_ptr_regs);
+    for (const auto& expression : body) {
+        const auto reg_info = expression->get_reg_info();
+        auto in_regs = transform_snippets_regs_to_idxs(reg_info.first);
+        auto out_regs = transform_snippets_regs_to_idxs(reg_info.second);
+        const auto& emitter = expression->get_emitter();
+        emitter->emit_code(in_regs, out_regs, vec_regs_pool, gp_regs_pool);
+    }
+
+    h->postamble();
+}
+
+jit_kernel_static_emitter::jit_kernel_static_emitter(dnnl::impl::cpu::aarch64::jit_generator* h, dnnl::impl::cpu::aarch64::cpu_isa_t isa,
+                                                     const ov::snippets::lowered::ExpressionPtr& expr)
+    : jit_kernel_emitter(h, isa, expr), reg_indexes_idx(Operand::X1) {
```

Comment:
Good point! I agree that KernelEmitter and other control-flow emitters like LoopBeginEmitter and LoopEndEmitter can be implemented as template class, as they mostly have common logics for different archs. Some arch-specific jit instructions should be implemented separately. Available gpr indices are different and usage of gprs could be different, for example x64 has specific logics to reuse gprs. But since the benefit is significant, I would vote for using template classes for these emitters.